Add `recordDiff` to ConfigPolicy and generator (#6095)
open-cluster-management/rhacm-docs: governance/config_policy_ctrl.adoc

| Optional
| Overrides `spec.object-templates[].complianceType` when comparing the manifest's metadata section to objects on the cluster ("musthave", "mustonlyhave"). Default is unset to not override `complianceType` for metadata.

| `spec.object-templates[].recordDiff`
| Optional

> | Specifies whether (and where) to log the difference between the object on the cluster and the `objectDefinition` in the policy. Set to "Log" to log the difference in the controller logs or "None" to not log the difference. Default is unset to not log the difference.

Reviewer: I monospaced the parameter values. Are the quotation marks required? 
```suggestion
| Specifies if and where to log the difference between the object on the cluster and the `objectDefinition` in the policy. Set to `Log` to log the difference in the controller logs or `None` to not log the difference. By default, this parameter is empty to not log the difference.
```

Author: Nope, monospacing without quotation marks works! Thanks!
